### DISCUSSION TITLE
Issues with async/await in packer.nvim

### DISCUSSION THREAD

**Discussion by c3n21** [wbthomason/packer.nvim#817] [Q&A] [answered]
Hello guys,
I've been working for a while on [this](https://github.com/wbthomason/packer.nvim/pull/370) PR for a while now but I always have some issues with the tests.

Locally I use Docker to do my own testing, and the tests seems to go well, but on GitHub workflow they fail miserably and I can't replicate the errors on my own machine.

Here is my Dockerfile
```
FROM archlinux:base-devel
SHELL ["/bin/bash", "-c"]
WORKDIR /setup
RUN pacman -Sy git neovim python --noconfirm
RUN git clone --depth 1 https://github.com/nvim-lua/plenary.nvim ~/.local/share/nvim/site/pack/vendor/start/plenary.nvim
WORKDIR /root/.cache/nvim/packer.nvim
RUN touch ceo 
RUN touch ceo1
RUN touch ceo2
WORKDIR /root/.local/share/nvim/site/pack/packer/start/packer.nvim/
COPY . ./
ENTRYPOINT make test
```
I've also tried to use the Dockerfile provided by packer.nvim but the tests run successfully.
I even tried to replicate the test environment using an Ubuntu container with the same dependencies and compiling on my own, but even then the tests are ok.

Makefile is almost the same, I've just added two commands to make running tests in Dockerfile easier
```
test:
	if [ ! -d ~/.local/share/nvim/site/pack/vendor/start/plenary.nvim ]; \
	then \
		git clone --depth 1 https://github.com/nvim-lua/plenary.nvim \
			~/.local/share/nvim/site/pack/vendor/start/plenary.nvim; \
		ln -f -s "$(pwd)" ~/.local/share/nvim/site/pack/vendor/start/packer.nvim; \
	fi; \
	nvim --headless --noplugin -u tests/minimal.vim \
		-c "PlenaryBustedDirectory tests/ {minimal_init = 'tests/minimal.vim'}"
run:
	docker build . -t neovim-stable:latest && docker run --rm -it --entrypoint bash neovim-stable:latest
run-test:
	docker build . -t neovim-stable:latest && docker run --rm neovim-stable:latest
```

It would be helpful if somebody could give me some tips to solve this issue or at least tell what I am doing wrong.
This is the error that I get on GitHub workflow

```
Packer testing  snapshot.create() create snapshot in '/home/runner/.cache/nvim/packer.nvim/test'	
            ...vendor/start/plenary.nvim/lua/plenary/async_lib/util.lua:326: Blocking on future timed out or was interrupted
            
            stack traceback:
            	...vendor/start/plenary.nvim/lua/plenary/async_lib/util.lua:326: in function <...vendor/start/plenary.nvim/lua/plenary/async_lib/util.lua:309>
```
Here is the branch I'm working on https://github.com/c3n21/packer.nvim/tree/master

Thanks in advance for your help :)

**Comment by wbthomason** [comment, marked as answer]
See response on #370.
